code/datums/components/mob_harvest.dm

```diff
@@ -0,0 +1,157 @@
+/**
+ * Harvesting component. Useful if you want to be able to harvest items from living mobs.
+ *
+ * Used currently on sheep.
+ */
+/datum/component/mob_harvest
+	///abstract item for managing the drops
+	var/obj/item/mob_harvest/mob_harvest
+	///item used to harvest
+	var/obj/item/harvest_tool = /obj/item/razor
+	///item used to reduce wait between items
+	var/obj/item/fed_item = /obj/item/food/grown/grass
+
+//harvest_type, produced_item_typepath and speedup_type are typepaths, not reference
+/datum/component/mob_harvest/Initialize(harvest_type = /obj/item/mob_harvest, speedup_type = fed_item)
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+	mob_harvest = new harvest_type(null, parent)
+
+/datum/component/mob_harvest/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/on_attackby)
+
+/datum/component/mob_harvest/UnregisterFromParent()
+	QDEL_NULL(mob_harvest)
+	UnregisterSignal(parent, list(COMSIG_PARENT_EXAMINE, COMSIG_PARENT_ATTACKBY))
+
+///signal called on parent being examined
+/datum/component/mob_harvest/proc/on_examine(datum/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+
+	if(mob_harvest.amount_ready < 1)
+		examine_list += span_notice("[parent] seems like they could use a bit more time.")
+	if(mob_harvest.amount_ready > 1)
+		examine_list += span_notice("[parent] looks like they can be harvested about [mob_harvest.amount_ready] times.")
+	if(mob_harvest.amount_ready == 1)
+		examine_list += span_notice("[parent] looks ready to be harvested.")
+
+///signal called on parent being attacked with an item
+/datum/component/mob_harvest/proc/on_attackby(datum/source, obj/item/used_item, mob/user)
+	SIGNAL_HANDLER
+
+	if(istype(used_item, harvest_tool))
+		mob_harvest.harvest_item(user)
+	if(istype(used_item, fed_item))
+		mob_harvest.remove_wait_time(user)
+		qdel(used_item)
+	return COMPONENT_NO_AFTERATTACK
+
+
+/**
+ * # Harvester item
+ *
+ * Abstract item that is held in nullspace and manages the time it takes to get a new item and creating said item
```

Comment:
this sounds like it should be a datum if it isn't actually meant to be in the world.

Reply:
From what mmm says above, they modelled udders. But udders had a good reason to be obj, because they were reagent holders. This doesn't